lib/memcached/memcached.rb

```diff
@@ -322,7 +322,7 @@ def set(key, value, ttl=@default_ttl, encode=true, flags=FLAGS)
       )
     rescue => e
       tries ||= 0
-      retry if e.instance_of?(ClientError) && !tries
```

Comment:
You're right, the `!tries` condition is wrong and would never be true.  Although, I'm not sure why you are adding `tries = 1 and` which seems like it would also make `tries == 0`, which seems like it would make it always retry.